drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,422 @@
+		err = i2c_reg_read_byte_dt(&config->bus, STMPE811_TSC_CTRL_REG, &tsc_ctrl);
+		if (err < 0) {
+			return err;
+		}
+
+		/* TOUCH interrupt + no touch detected in TSC_CTRL reg <=> release */
+		if (!(tsc_ctrl & STMPE811_TSC_CTRL_BIT_TOUCH_DET)) {
+			input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
+		}
+	}
+
+	return 0;
+}
+
+static void stmpe811_work_handler(struct k_work *work)
+{
+	struct stmpe811_data *data = CONTAINER_OF(work, struct stmpe811_data, processing_work);
+	const struct stmpe811_config *config = data->dev->config;
+
+	stmpe811_process(data->dev);
+	/**
+	 * Reschedule ISR if there was an interrupt triggered during handling (race condition).
+	 * IRQ is edge-triggered, so otherwise it would never be triggered again.
+	 */
+	if (gpio_pin_get_dt(&config->int_gpio)) {
+		k_work_submit(&data->processing_work);
+	}
+}
+
+static void stmpe811_interrupt_handler(const struct device *dev, struct gpio_callback *cb,
+				       uint32_t pins)
+{
+	struct stmpe811_data *data = CONTAINER_OF(cb, struct stmpe811_data, int_gpio_cb);
+
+	k_work_submit(&data->processing_work);
+}
+
+static int stmpe811_verify_chip_id(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	uint8_t buf[2];
+
+	i2c_burst_read_dt(&config->bus, STMPE811_CHP_ID_LSB_REG, buf, 2);
+
+	if (sys_get_be16(buf) != CHIP_ID) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stmpe811_init(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+	int err;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+
+	k_work_init(&data->processing_work, stmpe811_work_handler);
+
+	/* Verify CHIP_ID */
+	err = stmpe811_verify_chip_id(dev);
+	if (err) {
+		LOG_ERR("CHIP ID verification failed (%d)", err);
+		return err;
+	}
+
+	/* Initialize */
+	err = stmpe811_ts_init(dev);
+	if (err) {
+		LOG_ERR("Touch screen controller initialization failed (%d)", err);
+		return err;
+	}
+
+	/* Initialize GPIO interrupt */
+	if (!gpio_is_ready_dt(&config->int_gpio)) {
+		LOG_ERR("Interrupt GPIO controller device not ready");
+		return -ENODEV;
+	}
+
+	err = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+	if (err < 0) {
+		LOG_ERR("Could not configure interrupt GPIO pin (%d)", err);
+		return err;
+	}
+
+	err = gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+	if (err < 0) {
+		LOG_ERR("Could not configure GPIO interrupt (%d)", err);
+		return err;
+	}
+
+	gpio_init_callback(&data->int_gpio_cb, stmpe811_interrupt_handler,
+			   BIT(config->int_gpio.pin));
+	err = gpio_add_callback_dt(&config->int_gpio, &data->int_gpio_cb);
+	if (err < 0) {
+		LOG_ERR("Could not set GPIO callback (%d)", err);
+		return err;
+	}
+
+	/* Enable global interrupts */
+	err = i2c_reg_update_byte_dt(&config->bus, STMPE811_INT_CTRL_REG, BIT_MASK(8),
+				     STMPE811_INT_CTRL_BIT_GLOBAL_INT);
```

Comment:
Should `BIT_MASK(8)` have a define? It's used in other INT register related i2c transaction calls as well.